### PR TITLE
Managers FE operations

### PR DESCRIPTION
**Pull request by DANIDIA**

### CODE REVIEW

File: src/components/SelectFromQuery/SelectFromQuery.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
+import React, { useEffect, useState } from 'react';
+import { useLocation, useNavigate } from 'react-router-dom';
+
+export function SelectFromQuery({
```
Comment (RomanovYurii) [inline src/components/SelectFromQuery/SelectFromQuery.js:4]:
```suggestion
export function SelectDepartmentFromQuery({
```
?

Comment (RomanovYurii) [reply src/components/SelectFromQuery/SelectFromQuery.js:4]: And why is it "from query"?

Comment (RomanovYurii) [reply src/components/SelectFromQuery/SelectFromQuery.js:4]: Moreover, there is not even a single select in this component :D

File: src/views/PanelViews/ManagerViews/ManagerSingleView/ManagerSingleView.js [hunk 2]
```diff
@@ -1,36 +1,108 @@
...
 
 export function ManagerSingleView() {
     const navigate = useNavigate();
-    const editManagerDataOutlet = useOutlet();
+    const { managerID } = useParams();
     const { userData } = useContext(SessionContext);
+    const { managerData, setManagerData } = useContext(managerContext);
+    const [statusMessage, setStatusMessage] = useState('');
+
+    useEffect(() => {
```
Comment (RomanovYurii) [inline src/views/PanelViews/ManagerViews/ManagerSingleView/ManagerSingleView.js]: Don't use anonymous function call here, move it to a separate function

File: src/views/PanelViews/ManagerViews/components/ManagerFormComponent/ManagerFormComponent.js [hunk 3]
```diff
@@ -66,81 +61,66 @@ export function ManagerFormComponent({
...
 
-            <label>department: </label>
-            {formValues.departmentData?.name || 'no department'}
-            <button onClick={handleSelectDepartment}>Select department</button>
-            <br />
+            <SelectFromQuery
+                fieldName='Department:'
+                pathToSelect={`/${layoutsPaths.USER_PANEL}/${panelsPaths.DEPARTMENTS_PANEL}/${pathsInPanel.SEARCH}`}
+                initialValue={formValues.departmentData}
+                onRedirect={() =>
```
Comment (RomanovYurii) [inline src/views/PanelViews/ManagerViews/components/ManagerFormComponent/ManagerFormComponent.js]: Remove this callback to function in component body

File: src/views/PanelViews/helpers/userFormValidator.js [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+import { creationNecessaryFields } from '../ManagerViews/shared';
+import { areNecessaryFieldsEmpty } from './formDataHelpers';
+
+export function userFormValidator(formData, setStatusMessage) {
+    if (areNecessaryFieldsEmpty(formData, creationNecessaryFields)) {
+        setStatusMessage('Some of necessary fields are empty fields');
+        return false;
+    }
+
+    try {
```
Comment (RomanovYurii) [inline src/views/PanelViews/helpers/userFormValidator.js:12]: What part of this code may throw an error?

File: src/components/SelectFromQuery/SelectFromQuery.js (hunk 1 above)
Comment (DANIDIA) [reply src/components/SelectFromQuery/SelectFromQuery.js:4]: This component is used to changing department of user.  I think more sensible name can be  `changeDepartmentComponent`

File: src/views/PanelViews/helpers/userFormValidator.js (hunk 4 above)
Comment (DANIDIA) [reply src/views/PanelViews/helpers/userFormValidator.js:12]: `parsePhoneNumber` throws errors when gets alphabetic symbols

Comment (RomanovYurii) [reply src/views/PanelViews/helpers/userFormValidator.js:12]: ok

File: src/components/SelectFromQuery/SelectFromQuery.js (hunk 1 above)
Comment (RomanovYurii) [reply src/components/SelectFromQuery/SelectFromQuery.js:4]: Okay, let it be so

**Review by RomanovYurii — APPROVED** [review]

**Pull request merged**